servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -276,48 +275,53 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
     if (lastInode != null) {
       if (lastInode.isDirectory() && options.isDirectory()) {
         if (!lastInode.isPersisted() && options.isPersisted()) {
-          lastInode.setPersisted(true);
-          traversalResult.getPersisted().add(lastInode);
-          String ufsPath = mMountTable.resolve(getPath(lastInode)).toString();
-          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
-          ufs.mkdirs(ufsPath, false);
+          // The final path component already exists and is not persisted, so it should be added
+          // to the non-persisted Inodes of traversalResult.
+          traversalResult.getNonPersisted().add(lastInode);
```

Comment:
ahhh, I see. So that means <code>toPersistInodes</code> is always a superset of <code>traversalResult.getNonPersisted()</code>? This distinction needs to be commented somewhere.

Maybe we should just create another local variable similar <code>toPersistInodes</code> that represents the EXISTING inodes that need to be persisted, since <code>toPersistInodes</code> are ALL inodes that need to be persisted.